sprite.go

```diff
@@ -27,18 +27,45 @@ func (s *SpriteSheet) Release() {
 }
 
 // DrawSprite draws a sprite from the sprite sheet at the given position.
-func (s *SpriteSheet) DrawSprite(col, row uint, pos Position, flip bool) {
+func (s *SpriteSheet) DrawSprite(col, row uint, pos Position, scale float32, flip bool) {
 	src := Rectangle{
 		Pos: Position{
 			int(s.frameSize.W) * int(col),
 			int(s.frameSize.H) * int(row),
 		},
 		Size: s.frameSize,
 	}
+	scaledWidth := int(float32(s.frameSize.W) * scale)
+	scaledHeight := int(float32(s.frameSize.H) * scale)
+
+	dest := Rectangle{
+		Pos: Position{
+			X: pos.X,
+			Y: pos.Y - (scaledHeight - s.frameSize.H),
+		},
+		Size: Size{
+			scaledWidth,
+			scaledHeight,
+		},
+	}
+
+	origin := Position{
+		X: 0, //TODO: with lower scales sprites are rendered on the left side
```

Comment:
Example

<img width="734" alt="image" src="https://github.com/user-attachments/assets/c7926233-e848-4b26-940a-3b59b2ae2678">

Reply:
That's definitely not what we want. As it is not using the world resolution of 320x200 pixels.